tests/brokermetadata/BrokerMetadataTest.cc

```diff
@@ -0,0 +1,58 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+// Run `docker-compose up -d` to set up the test environment for this test.
+#include <gtest/gtest.h>
+#include <pulsar/Client.h>
+
+using namespace pulsar;
+
+#ifndef TEST_ROOT_PATH
+#define TEST_ROOT_PATH "."
+#endif
+
+static const std::string gKeyPath = std::string(TEST_ROOT_PATH) + "/../test-conf/cpp_credentials_file.json";
+static std::string gClientId;
+static std::string gClientSecret;
+static ParamMap gCommonParams;
+
+TEST(BrokerMetadataTest, testConsumeSuccess) {
+    ClientConfiguration conf;
+    Client client{"pulsar://localhost:6650", conf};
+    Producer producer;
+    Result producerResult = client.createProducer("persistent://public/default/testConsumeSuccess", producer);
+    ASSERT_EQ(producerResult, ResultOk);
+    Consumer consumer;
+    Result consumerResult =
+        client.subscribe("persistent://public/default/testConsumeSuccess", "testConsumeSuccess", consumer);
+    ASSERT_EQ(consumerResult, ResultOk);
+    const Message& msg = MessageBuilder().setContent("testConsumeSuccess").build();
```

Comment:
Don't use a reference to a temporary value.

Just use value as the returned value because the copy overhead of `Message` is cheap since the only field of `Message` is a `std::shared_ptr`. Even if it's in a critical path, the overhead of copying could still be optimized.

```c++
const auto msg = MessageBuilder().setContent("testConsumeSuccess").build();
```

Reply:
@BewareMyPower Fixed